homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -0,0 +1,162 @@
+"""Conversation support for the Google Generative AI Conversation integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Literal
+
+from google.api_core.exceptions import ClientError
+import google.generativeai as genai
+import google.generativeai.types as genai_types
+
+from homeassistant.components import assist_pipeline, conversation
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import TemplateError
+from homeassistant.helpers import intent, template
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import ulid
+
+from .const import (
+    CONF_CHAT_MODEL,
+    CONF_MAX_TOKENS,
+    CONF_PROMPT,
+    CONF_TEMPERATURE,
+    CONF_TOP_K,
+    CONF_TOP_P,
+    DEFAULT_CHAT_MODEL,
+    DEFAULT_MAX_TOKENS,
+    DEFAULT_PROMPT,
+    DEFAULT_TEMPERATURE,
+    DEFAULT_TOP_K,
+    DEFAULT_TOP_P,
+)
+
+_LOGGER = logging.getLogger(__name__)
```

Comment:
For consistency with OpenAI move this into const.py?

Reply:
I didn't originally also copy this because the log messages would be weird:
```
ERROR:homeassistant.components.google_generative_ai_conversation.const:Error rendering prompt: TemplateSyntaxError: Unexpected end of template. Jinja was looking for the following tags: 'endif'. The innermost block that needs to be closed is 'if'.
```

I went ahead and copied it but updated to use `__package__` instead of `__name__` which I think was intended and will include in my openai cleanup PR.